### PR TITLE
Update LLVM

### PR DESCRIPTION
**Pull request by AtariDreams**
Update LLVM to 15.0.2.

### CODE REVIEW

**Comment by frederick-vs-ja** [comment]
This is not exactly a duplicate of #2976, but I guess we should continue works in that PR.

**Comment by CaseyCarter** [comment]
@AtariDreams I appreciate the enthusiasm, but (1) we have an open LLVM update PR in progress already (#2976) and (2) we always track `main` instead of a release branch. We'll hopefully get #2976 merged soon - I have a workaround for the test failures, I just need to file a couple of compiler bugs. After that, we'll need another update PR for the tip of `main` so we can start the process again.

TLDR: don't be surprised if nobody touches this for a while, and if we push an update for the submodule reference after #2976 merges.

**Comment by frederick-vs-ja** [comment]
### Failed tests failed due to unimplemented features
[P2286R8](https://wg21.link/p2286r8), tracked in #2919
* `std/utilities/format/format.formattable/concept.formattable.compile.pass.cpp` (the `std::formattable` concept)

[P0482R6](https://wg21.link/p0482r6), tracked in #2207, awaiting UCRT
* `std/depr/depr.c.headers/uchar_h.compile.pass.cpp` (`c8rtomb` and `mbrtoc8` in `<uchar.h>`)
* `std/strings/c.strings/cuchar.compile.pass.cpp` (same functions in `<cuchar>`)

These tests should be skipped in `skipped_tests.txt` and marked with `FAIL` in `expected_results.txt`.

### Failed tests due to unknown reasons
* `std/utilities/meta/meta.unary/dependent_return_type.compile.pass.cpp`
* `std/ranges/range.adaptors/range.take.while/begin.pass.cpp`
* `std/ranges/range.adaptors/range.take.while/end.pass.cpp`
* `std/ranges/range.factories/range.istream.view/begin.pass.cpp`
* `std/ranges/range.factories/range.istream.view/cpo.pass.cpp`
* `std/ranges/range.factories/range.istream.view/iterator/compare.pass.cpp`
* `std/utilities/variant/variant.variant/implicit_ctad.pass.cpp`

Most of these tests passed in 17.3 locally, perhaps they failed due to warnings. MSVC STL might be fixed to pass them.

### Unexpectedly passed tests
Passed because libc++ has implemented [P0591R4](https://wg21.link/p0591r4) and relative DRs.

We should just remove these stuffs:
https://github.com/microsoft/STL/blob/1fe4d4319b74d6d0738420b549fdfdacb304a647/tests/libcxx/expected_results.txt#L525-L533

https://github.com/microsoft/STL/blob/1fe4d4319b74d6d0738420b549fdfdacb304a647/tests/libcxx/skipped_tests.txt#L526-L534

File: tests/libcxx/skipped_tests.txt [hunk 1]
```diff
@@ -523,20 +523,17 @@ iterators\iterator.primitives\iterator.operations\robust_against_adl.pass.cpp
...
-utilities\allocator.adaptor\allocator.adaptor.members\construct_pair_values.pass.cpp
-utilities\allocator.adaptor\allocator.adaptor.members\construct_type.pass.cpp
-
 # Tests expect __cpp_lib_ranges to have the old value 201811L for P0896R4; we define the C++20 value 201911L for P1716R3.
 language.support\support.limits\support.limits.general\algorithm.version.compile.pass.cpp
 language.support\support.limits\support.limits.general\functional.version.compile.pass.cpp
 language.support\support.limits\support.limits.general\iterator.version.compile.pass.cpp
 
+# Did not implement P2286R8 yet
+std/utilities/format/format.formattable/concept.formattable.compile.pass.cpp
```
Comment (CaseyCarter) [inline tests/libcxx/skipped_tests.txt:532]: Entries in this file don't have the leading `std` and use `\` path separators.

File: tests/libcxx/expected_results.txt [hunk 2]
```diff
@@ -522,14 +521,11 @@ std/iterators/iterator.primitives/iterator.operations/robust_against_adl.pass.cp
...
 
-# Uses-allocator class constructor wants non-const allocator ref and mismatched piecewise_construct's value category
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct.pass.cpp FAIL
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct_pair.pass.cpp FAIL
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct_pair_const_lvalue_pair.pass.cpp FAIL
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct_pair_piecewise.pass.cpp FAIL
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct_pair_rvalue.pass.cpp FAIL
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct_pair_values.pass.cpp FAIL
-std/utilities/allocator.adaptor/allocator.adaptor.members/construct_type.pass.cpp FAIL
+# Did not implement P2286R8 yet
```
Comment (CaseyCarter) [inline tests/libcxx/expected_results.txt:524]: For clarity, it would be nice to specify whether _we_ don't implement p2286r8 or libc++ doesn't. (Also on 527. Be sure to mirror any changes in `skipped_tests.txt`.)

File: tests/libcxx/expected_results.txt [hunk 3]
```diff
@@ -257,7 +257,6 @@ std/ranges/range.utility/range.subrange/structured_bindings.pass.cpp:0 SKIPPED
 
 # DevCom-1436243 constexpr new initialized array
 std/utilities/smartptr/unique.ptr/unique.ptr.class/unique.ptr.modifiers/reset_self.pass.cpp:0 FAIL
-std/utilities/smartptr/unique.ptr/unique.ptr.create/make_unique.array.pass.cpp:0 FAIL
```
Comment (CaseyCarter) [inline tests/libcxx/expected_results.txt:260]: I don't see a corresponding change to `skipped_tests.txt`.

File: tests/libcxx/expected_results.txt [hunk 4]
```diff
@@ -875,9 +871,7 @@ std/utilities/format/format.functions/unicode.pass.cpp:0 FAIL
 #Not yet analyzed. "constexpr evaluation hit maximum step limit; possible infinite loop?"
 std/utilities/template.bitset/bitset.members/left_shift_eq.pass.cpp FAIL
 std/utilities/template.bitset/bitset.members/op_and_eq.pass.cpp FAIL
-std/utilities/template.bitset/bitset.members/op_or_eq.pass.cpp FAIL
 std/utilities/template.bitset/bitset.members/right_shift_eq.pass.cpp FAIL
-std/utilities/template.bitset/bitset.members/to_string.pass.cpp:1 FAIL
```
Comment (CaseyCarter) [inline tests/libcxx/expected_results.txt:880]: Ditto "missing a corresponding change to `skipped_tests.txt`"

File: tests/libcxx/expected_results.txt [hunk 5]
```diff
@@ -1044,8 +1038,6 @@ std/utilities/function.objects/refwrap/refwrap.const/type_conv_ctor.pass.cpp:0 F
 std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.create/allocate_shared.array.unbounded.pass.cpp FAIL
 std/utilities/meta/meta.logical/conjunction.compile.pass.cpp FAIL
 std/utilities/meta/meta.logical/disjunction.compile.pass.cpp FAIL
-std/utilities/meta/meta.unary/meta.unary.prop/is_nothrow_copy_assignable.pass.cpp:0 FAIL
-std/utilities/meta/meta.unary/meta.unary.prop/is_nothrow_move_assignable.pass.cpp:0 FAIL
```
Comment (CaseyCarter) [inline tests/libcxx/expected_results.txt:1048]: And again "missing a corresponding change to `skipped_tests.txt`"

**Review by CaseyCarter — CHANGES_REQUESTED** [review]
We're not interested in updating our LLVM submodule reference to LLVM-15.0.2. We do not track any release branch, we track `main`. We _would_ be interested in updating our submodule reference to something nearer the current tip of `main`.
